test/primus.test.js

```diff
@@ -104,6 +104,22 @@ describe('Primus', function () {
 
     throw new Error('I should have throwed above');
   });
+  
+  it('accepts a third-party spark id generator', function () {
+    function testIdGenerator() {
+      return 'test_'+(Math.random()+1).toString(36).substring(7);
```

Comment:
Can't we simply return a dummy string like `'foo'` and create only a spark below?
I think it's sufficient as all we need to test is that the function is invoked and the spark id is set to the return value of this function.